### PR TITLE
Improving logging while validation dataframes

### LINKED ISSUE #12: on failure, optionally provide data on errors (by row & reason)
at the moment data issues can either cause an Exception or for data to be dropped, depending on args used.

Having a way to get data back on where and why errors occurred across a whole DF would be useful

Thanks!

### PR DESCRIPTION
**Pull request by wesselhuising**
The goal of the PR is to improve the per row logging in case the validation fails.

### CODE REVIEW

**Comment by xaviernogueira** [comment]
@wesselhuising 
That said, now that I am thinking about it, for `filter()` the options could be a touch different. With silently filtering as the default, and logging each failed row as the option.

**Comment by wesselhuising** [comment]
> @wesselhuising That said, now that I am thinking about it, for `filter()` the options could be a touch different. With silently filtering as the default, and logging each failed row as the option.

Yeah so I think we just need the verbose flag for both methods, so the old situation haha.

**Comment by xaviernogueira** [comment]
@wesselhuising lol! Good point, but with the row based logging that is.

File: src/pandantic/validators/pandas.py [hunk 1]
```diff
@@ -22,26 +22,28 @@ def __init__(self, schema: SchemaTypes):
...
         queue: Optional[Queue] = None,
-        verbose: bool = True,
     ) -> pd.DataFrame:
         """Validate a DataFrame using the schema defined in the Pydantic model.
 
         Args:
             dataframe (pd.DataFrame): The DataFrame to validate.
-            errors (str, optional): How to handle validation errors. Defaults to "raise".
-            context (Optional[dict[str, Any], None], optional): The context to use for validation.
+            errors (Literal["skip", "raise", "log"], optional): How to handle validation errors. Defaults to "raise".
```
Comment (xaviernogueira) [inline src/pandantic/validators/pandas.py:36]: Perhaps add a brief explanation that "skip"/"log" effectively filter?

**Review by xaviernogueira — APPROVED** [review]
Not too much to say, code wise it looks good! I like the use of `typing.Literal`.  I think the docstring can make the filtering logic a bit more clear, but that is a bit of a nit and does not need to hold up merge.

There was a minor issue with the logic where if you set `errors="log"` it would not return the filtered version, but I quickly fixed it (see my commits). I also updated the pandas plugin to work with thew new pattern.

LGTM 🚀

File: src/pandantic/validators/pandas.py (hunk 1 above)
Comment (xaviernogueira) [reply src/pandantic/validators/pandas.py:36]: I'll just add it real quick!

**Pull request merged**